cmd/cut-release/main.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/plaid/go-envvar/envvar"
+)
+
+type envVars struct {
+	// Version is the new release version to use
+	Version string `envvar:"VERSION" default:"development"`
+}
+
+func main() {
+	env := envVars{}
+	if err := envvar.Parse(&env); err != nil {
+		log.Fatal(err)
+	}
+
+	updateHardCodedVersions(env.Version)
+
+	generateTypescriptClientDocs()
+}
+
+func generateTypescriptClientDocs() {
+	// Run `yarn install` to make sure `TypeDoc` dep is installed
+	cmd := exec.Command("yarn", "install")
+	cmd.Dir = "rpc/clients/typescript"
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Run `yarn docs:md` to generate MD docs
+	cmd = exec.Command("yarn", "docs:md")
+	cmd.Dir = "rpc/clients/typescript"
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Update SUMMARY.md
+	tsClientSummaryPath := "docs/json_rpc_clients/typescript/SUMMARY.md"
+	dat, err := ioutil.ReadFile(tsClientSummaryPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Modify the paths to be prefixed with `json_rpc_clients/typescript`
+	modifiedDat := strings.Replace(string(dat), "](", "](json_rpc_clients/typescript/", -1)
+	modifiedDat = strings.Replace(modifiedDat, "](json_rpc_clients/typescript/)", "]()", -1)
+	finalTsClientSummary := strings.Replace(modifiedDat, "* [", "  * [", -1)
+
+	// Replace the summary content nested under `Typescript client`
+	mainSummaryPath := "docs/SUMMARY.md"
+	dat, err = ioutil.ReadFile(mainSummaryPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	finalSummaryLines := []string{}
+	lines := strings.Split(string(dat), "\n")
+	isCutting := false
+	for _, l := range lines {
```

Comment:
I think this could use a comment explaining the significance of the `"## Contributing"` and `"[Typescript client]"` lines.